File: testFiles/calculator.c
int stackData[1000 * 2];
int stackNo[1];
int initStack() {
    int ans = stackNo[0];
    stackNo[0] = stackNo[0] + 1;
    stackData[1000 * ans] = 0;
    return ans;
}
void push(int stack, int tar) {
    int length = stackData[stack * 1000];
    if (stack >= 2 || stack < 0) {
        return;
    }
    stackData[stack * 1000 + length + 1] = tar;
    stackData[stack * 1000] = length + 1;
}
int pop(int stack) {
    if (stack >= 2) {
        return 0;
    }
    if (stack < 0) {
        return 0;
    }
    int length = stackData[stack * 1000];
    if (length <= 0) {
        return 0;
    }
    stackData[stack * 1000] = length - 1;
    return stackData[stack * 1000 + length];
}
int stackEmpty(int stack) {
    if (stack >= 2) {
        return 1;
    }
    if (stack < 0) {
        return 1;
    }
    int length = stackData[stack * 1000];
    if (length == 0) {
        return 1;
    }
    return 0;
}
int getTop(int stack) {
    if (stack >= 2) {
        return 0;
    }
    if (stack < 0) {
        return 0;
    }
    int length = stackData[stack * 1000];
    if (length <= 0) {
        return 0;
    }
    return stackData[stack * 1000 + length];
}

char str[] = "5+6*(3+2)-1";

int Priority(int s)
{
    if (s == '(')
    {
        return 3;
    }
    if (s == '*' || s == '/') {
        return 2;
    }
    if (s == '+' || s == '-') {
        return 1;
    }
    return 0;
}

int main()
{
    stackNo[0] = 0;
    int num = initStack();
    int opt = initStack();
    int i = 0;
    int tmp = 0;
    int j;
    int strlen_t = strlen(str);

    while (stackEmpty(opt) != 1 || i < strlen_t)
    {
        if(i < strlen_t){

        if(str[i] >= '0' && str[i] <= '9')
        {
            tmp = tmp * 10 + (int)(str[i]);
            i = i + 1;
            if (i >= strlen_t){
                push(num, tmp);
                tmp = 0;
            }
            else {

            if(str[i] >= '0' && str[i] <= '9') {
                continue;
            }
            else {
                push(num, tmp);
                tmp = 0;
            }
            }
        }
        else
        {
            if(stackEmpty(opt) || Priority(str[i]) > Priority(getTop(opt)))
            {
                push(opt, str[i]);
                i = i + 1;
                continue;
            }
            if (getTop(opt) == '(' && str[i] != ')') {
                push(opt, str[i]);
                i = i + 1;
                continue;
            }
            if(getTop(opt) == '(' && str[i] == ')')
            {
                pop(opt);
                i = i + 1;
                continue;
            }
            int ok = 0;
            if (stackEmpty(opt) != 1 && str[i] == '\0') {
                ok = 1;
            }
            if(str[i] == ')' && getTop(opt) != '(') {
                ok = 1;
            }
            if (Priority(str[i]) <= Priority(getTop(opt))) {
                ok = 1;
            }
            if (ok) {
                int opt_now = pop(opt);
                if (opt_now ==  '+') {
                    push(num, pop(num) + pop(num));
                }
                if (opt_now ==  '-') {
                    j = pop(num);
                    push(num, pop(num) - j);
                }
                if (opt_now ==  '*') {
                    push(num, pop(num) * pop(num));
                }
                if (opt_now ==  '/') {
                    j = pop(num);
                    push(num, pop(num) / j);
                }
            }
        }
        }else{
            int ok = 0;
            if (stackEmpty(opt) != 1 && i == strlen_t) {
                ok = 1;
            }
            if (ok) {
                int opt_now = pop(opt);
                if (opt_now ==  '+') {
                    push(num, pop(num) + pop(num));
                }
                if (opt_now ==  '-') {
                    j = pop(num);
                    push(num, pop(num) - j);
                }
                if (opt_now ==  '*') {
                    push(num, pop(num) * pop(num));
                }
                if (opt_now ==  '/') {
                    j = pop(num);
                    push(num, pop(num) / j);
                }
            }
        }
    }

    printf("%d\n", pop(num));
    return 0;
}

File: output/calculator.py
data = [0] * 1000*2
count = [0] * 1
orderList = [0, 1, 2, 3]
calStr = "1+3*6/2-2"


def init():
	ans = count[0]
	count[0]=count[0]+1
	data[1000*ans]=0
	return ans


def order(opt):
	if opt == '(': 
		return orderList[3]
	if opt == '*' or opt == '/': 
		return orderList[2]
	if opt == '+' or opt == '-': 
		return orderList[1]
	return orderList[0]


def ifEmpty(stack):
	len = data[stack*1000]
	if stack<0: 
		return 1
	if stack>=2: 
		return 1
	if len == 0: 
		return 1
	return 0


def push(stack, ele):
	len = data[stack*1000]
	if stack>=2 or stack<0: 
		return
	data[stack*1000+len+1]=ele
	data[stack*1000]=len+1


def pop(stack):
	len = data[stack*1000]
	if stack<0: 
		return 0
	if stack>=2: 
		return 0
	if len<=0: 
		return 0
	data[stack*1000]=len-1
	return data[stack*1000+len]


def getStackTop(stack):
	len = data[stack*1000]
	if stack<0: 
		return 0
	if stack>=2: 
		return 0
	if len<=0: 
		return 0
	return data[stack*1000+len]


def main():
	count[0]=0
	num = init()
	operator = init()
	i = 0
	tmp = 0
	
	strlen_t = len(calStr)
	print("%s = " % (calStr), end ='')
	while ifEmpty(operator) == 0 or i<strlen_t:
		if i<strlen_t: 
			if calStr[i]>='0' and calStr[i]<='9': 
				tmp=tmp*10+int(calStr[i])
				i=i+1
				if i>=strlen_t: 
					push(num, tmp)
					tmp=0
				else:
					if calStr[i]>='0' and calStr[i]<='9': 
						continue
					else:
						push(num, tmp)
						tmp=0
			else:
				if ifEmpty(operator) or order(calStr[i])>order(getStackTop(operator)): 
					push(operator, calStr[i])
					i=i+1
					continue
				if getStackTop(operator) == '(' and calStr[i] != ')': 
					push(operator, calStr[i])
					i=i+1
					continue
				if getStackTop(operator) == '(' and calStr[i] == ')': 
					pop(operator)
					i=i+1
					continue
				if (ifEmpty(operator) != 1 and calStr[i] == '\0') or (calStr[i] == ')' and getStackTop(operator) != '(') or (order(calStr[i])<=order(getStackTop(operator))): 
					operatorNow = pop(operator)
					if operatorNow == '+': 
						push(num, pop(num)+pop(num))
					if operatorNow == '-': 
						j=pop(num)
						push(num, pop(num)-j)
					if operatorNow == '*': 
						push(num, pop(num)*pop(num))
					if operatorNow == '/': 
						j=pop(num)
						push(num, pop(num)/j)
		else:
			if ifEmpty(operator) == 0 and i == strlen_t: 
				operatorNow = pop(operator)
				if operatorNow == '+': 
					push(num, pop(num)+pop(num))
				if operatorNow == '-': 
					j=pop(num)
					push(num, pop(num)-j)
				if operatorNow == '*': 
					push(num, pop(num)*pop(num))
				if operatorNow == '/': 
					j=pop(num)
					push(num, pop(num)/j)
	print("%d\n"% ((pop(num))), end = '')
	return 0


if __name__ == '__main__': 
	main()
